src/cpp/rtps/builtin/discovery/endpoint/EDPSimple.cpp

```diff
@@ -471,7 +471,7 @@ bool EDPSimple::processLocalReaderProxyData(RTPSReader* local_reader, ReaderProx
     if(writer->first != nullptr)
     {
         // TODO(Ricardo) Write a getCdrSerializedPayload for ReaderProxyData.
-        CacheChange_t* change = writer->first->new_change([]() -> uint32_t {return DISCOVERY_SUBSCRIPTION_DATA_MAX_SIZE;},
+        CacheChange_t* change = writer->first->new_change([this]() -> uint32_t {return mp_PDP->builtin_attributes().readerPayloadSize;},
```

Comment:
This would be `writerPayloadSize`, as we are adding the change to the built-in writer.